src/com/google/enterprise/adaptor/database/DatabaseAdaptor.java

```diff
@@ -697,18 +703,20 @@ private Acl getAcl(Connection conn, String uniqueId) throws SQLException {
     try (PreparedStatement stmt = getAclFromDb(conn, uniqueId);
         ResultSet rs = stmt.executeQuery()) {
       log.finer("got acl");
-      return buildAcl(rs, aclPrincipalDelimiter, aclNamespace);
+      boolean hasResult = rs.next();
+      if (!hasResult) {
+        // empty Acl ensures adaptor will mark this document as secure
+        return Acl.EMPTY;
```

Comment:
Why move this code from buildAcl?

Reply:
Because the other call to buildAcl from getDocContent using the content SQL doesn't ever want to call rs.next(). We could check rsNext at the top and bottom of the loop in buildAcl, which is messier and another double-check, but it's in one method.